insights/core/spec_factory.py

```diff
@@ -677,6 +678,54 @@ def __call__(self, broker):
         raise ContentException("[%s] didn't match." % ', '.join(self.patterns))
 
 
+class recent_files(object):
```

Comment:
May I know if the `last_modify_hours` is a key filter for this spec? Per my understanding according to our discussion, TBH, enhancing the existing `listdir`, e.g. adding a few more arguments `sort, key, type, reverse, num`, looks like a better approach than adding this new one from my perspective.

E.g. 

The usage of the update `listdir` in your case would be:

```python
eap_json_reports = head(
    listdir("/var/tmp/insights-runtimes/uploads/", 
        sort=True, type=os.path.isfile, key=os.path.getmtime, reverse=True),
    num=30
)
```

the update to the `listdir` and `head` will also be quite simple:
```python
class listdir(object):
    def __init__(self, path, context=None, ignore=None, deps=[], sort=False, type=None, key=None, reverse=False):
...
        self.sort, self.type, self.key, self.reverse, self.num = sort, type, key, reverse


    def __call__(self, broker):
...
        try:
            result = os.listdir(p)
            if sort:
                result = sorted(filter(self.type, result), key=self.key, reverse=self.reverse)
...

class head(object):
...
    def __call__(self, lst):
        c = lst[self.dep]
        if lst:
            return c[:self.num]
```

Reply:
last_modify_hours is key as currently the requirement is to  collect files last 24 hours updated.  So if update other specs, need to add this parameter and that will make existing (head/listdir/glob_file) specs more complex. I'd like to add new one .

Reply:
As this requirement is particular, I suggest not adding such a common helper but implementing it in an individual datasource.